app/controllers/v0/institutions_controller.rb

```diff
@@ -88,5 +88,14 @@ def facets
       }
     end
     # rubocop:enable AbcSize
+
+    # Embed search result counts as a list of hashes with "name"/"count"
+    # keys so that open-ended strings such as country names do not
+    # get interpreted/mutated as JSON keys.
+    def embed(group_counts)
+      group_counts.each_with_object([]) do |(k, v), array|
+        array << { name: k, count: v }
+      end
```

Comment:
In that case it would be a lot more readable if we used the active record methods than these custom query methods in my opinion. I'll have to demonstrate with an example later, but if active record is used to do the group directly, you can actually build the hash you are building in the embed method here automatically (no need for the additional iterating over each object).